app/interactors/comments/destroy.rb

```diff
@@ -0,0 +1,15 @@
+module Comments
+  class Destroy
+    include Interactor
+
+    delegate :comment, to: :context
+
+    def call
+      if comment.destroy
+        context.status = :success
+      else
+        context.fail!(error: "Failed to delete comment")
+      end
```

Comment:
Interactor will have success context always while we don't fail that
```suggestion
      context.fail!(error: "Failed to delete comment") unless comment.destroy
```